### PR TITLE
Add operatorId filter to ArrivalByTime widget

### PR DESCRIPTION
**Pull request by i5x64BIT**
https://github.com/hasadna/open-bus-map-search/assets/62185136/630c92ef-1056-4fbc-9786-9e289eecbd4a

#148
Added a operatorId filter in Arrival widget

### CODE REVIEW

**Comment by i5x64BIT** [comment]
I updated the title too, it looks good but I could not verify. I couldn't get the text to update.

**Comment by i5x64BIT** [comment]
I did a --amend commit, droped the title one on force pushed it. Hope it didn't mess anything up

**Comment by NoamGaash** [comment]
force push are allowed here, but there's no need to amend commits because we squash merge our PRs

File: src/pages/dashboard/ArrivalByTimeChart/ArrivalByTimeChart.tsx [hunk 1]
```diff
@@ -35,7 +37,16 @@ export default function ArrivalByTimeChart({
     gtfs_route_date: string
     gtfs_route_hour: string
   }[]
+  operatorId: string
 }) {
+
+  const filteredData = useMemo(
+    () => data.filter((item) => operatorId && item.id === operatorId),
```
Comment (NoamGaash) [inline src/pages/dashboard/ArrivalByTimeChart/ArrivalByTimeChart.tsx:44]: why not
```suggestion
    () => data.filter((item) => !operatorId || item.id === operatorId),
```
?

**Review by NoamGaash — APPROVED** [review]
Looks good

File: src/pages/dashboard/ArrivalByTimeChart/ArrivalByTimeChart.tsx (hunk 1 above)
Comment (i5x64BIT) [reply src/pages/dashboard/ArrivalByTimeChart/ArrivalByTimeChart.tsx:44]: Damn that's a smart one, haven't thought of that.
Can I still update the pr?

Comment (NoamGaash) [reply src/pages/dashboard/ArrivalByTimeChart/ArrivalByTimeChart.tsx:44]: sure

Comment (NoamGaash) [reply src/pages/dashboard/ArrivalByTimeChart/ArrivalByTimeChart.tsx:44]: in fact, you'll need to update it anyway as the linter is failing

**Comment by i5x64BIT** [comment]
@NoamGaash I've tried to `yarn lint` before commiting and received this error. any advice?
Screenshot 2023-11-15 at 9 13 02

**Comment by NoamGaash** [comment]
that's just a warning,
seems like the linter is still running on your machine
you can take a look at the linter output of the github runner - https://github.com/hasadna/open-bus-map-search/actions/runs/6859008868/job/18652201225?pr=204
```
Run yarn lint

/home/runner/work/open-bus-map-search/open-bus-map-search/src/pages/dashboard/ArrivalByTimeChart/ArrivalByTimeChart.tsx
Error:   [4](https://github.com/hasadna/open-bus-map-search/actions/runs/6859008868/job/18652201225?pr=204#step:5:5)2:1  error  Delete `⏎`  prettier/prettier

/home/runner/work/open-bus-map-search/open-bus-map-search/src/pages/dashboard/DashboardPage.tsx
Error:   16[5](https://github.com/hasadna/open-bus-map-search/actions/runs/6859008868/job/18652201225?pr=204#step:5:6):34  error  Replace `·data={convertToGraphCompatibleStruct(graphData)}·operatorId={operatorId}` with `⏎················data={convertToGraphCompatibleStruct(graphData)}⏎················operatorId={operatorId}⏎·············`  prettier/prettier

✖ 2 problems (2 errors, 0 warnings)
  2 errors and 0 warnings potentially fixable with the `--fix` option.

Error: Process completed with exit code 1.
```

**Comment by i5x64BIT** [comment]
@NoamGaash Just know, what should I run before commiting?

**Comment by NoamGaash** [comment]
@i5x64BIT build, lint and test

**Pull request merged**